### ISSUE TITLE
oauth2 example fails with a local test server

### ISSUE THREAD

**Issue by chmorgan** [jaredhanson/passport-google-oauth#38]
Set express back to 3.x to get things working, updated the url like:
-    callbackURL: "http://127.0.0.1:3000/auth/google/callback"
-    callbackURL: "http://192.168.1.10:3000/auth/google/callback"

plugged in my client and secret values, and google is reporting an error:

"
400. That’s an error.

Error: invalid_request

device_id and device_name are required for private IP: http://192.168.1.10:3000/auth/google/callback

Learn more

Request Details
That’s all we know.
"

**Comment by debashish-ghosh** [comment]
I cannot use the hosts solution because there are about 20 different systems which will be using the site.

**Comment by aagam29** [comment]
I'm working with a container , facing the same issue

**Comment by matteoopenf** [comment]
found a solution?

**Comment by aagam29** [comment]
Yes. Change ip with domain name. And add it in google cloud console

**Comment by matteoopenf** [comment]
many thanks

**Comment by subgero** [comment]
> Yes. Change ip with domain name. And add it in google cloud console

Could you elaborate the solution please.

**Comment by tananga** [comment]
xD This issue is from 2014 and still not solved

**Comment by huyprowow** [comment]
could you try this: https://stackoverflow.com/questions/24736168/error-invalid-request-device-id-and-device-name-are-required-for-private-ip
